cg/meta/upload/fohm/fohm.py

```diff
@@ -0,0 +1,199 @@
+        """
+        Hardlink samples rawdata files to fohm delivery folder
+        """
+        for sample_id in self.aggregation_dataframe["internal_id"]:
+            sample_obj: models.Sample = self.status_db.sample(sample_id)
+            bundle_name = sample_obj.links[0].family.internal_id
+            version_obj: Version = self.housekeeper_api.last_version(bundle=bundle_name)
+            files = self.housekeeper_api.files(version=version_obj.id, tags=[sample_id]).all()
+            for file in files:
+                shutil.copy(file.full_path, Path(self.daily_rawdata_path))
+
+    def reaggregate_reports(self):
+        unique_regionlabs = list(self.aggregation_dataframe["region_lab"].unique())
+        for region_lab in unique_regionlabs:
+            pangolin_df = self.pangolin_dataframe[
+                self.aggregation_dataframe["region_lab"] == region_lab
+            ]
+            pangolin_df.to_csv(
+                self.daily_rawdata_path
+                / f"{region_lab}_{self.current_datestr}_pangolin_classification_format3.txt",
+                sep="\t",
+                index=False,
+            )
+            report_df = self.reports_dataframe[
+                self.aggregation_dataframe["region_lab"] == region_lab
+            ]
+            report_df.to_csv(
+                self.daily_report_path / f"{region_lab}_{self.current_datestr}_komplettering.csv",
+                sep=",",
+                index=False,
+            )
+
+    def assemble_fohm_delivery(self, cases: List[str]) -> None:
+        """Rearrange data and put in delivery dir"""
+        self._cases_to_aggregate = cases
+        self.create_daily_delivery_folders()
+        self.append_metadata_to_aggregation_df()
+        self.reaggregate_reports()
+        self.link_sample_rawdata()
+
+    def send_mail_reports(self, date: Optional[str] = None):
+        if date:
+            self._current_datestr = date
+        for file in self.daily_report_path.iterdir():
+            send_mail(
+                EmailInfo(
+                    receiver_email="clinicaldelivery@scilifelab.se",
+                    sender_email="clinical.genomics@hasta.scilifelab.se",
+                    smtp_server="hasta.scilifelab.se",
+                    subject="Komplettering",
+                    message=" ",
+                    file=file,
+                )
+            )
+
+    def update_upload_started_at(self, case_id: str):
+        """Update timestamp for cases which started being processed as batch"""
+        case_obj: models.Family = self.status_db.family(case_id)
+        case_obj.analyses[0].upload_started_at = dt.datetime.now()
+        self.status_db.commit()
+
+    def update_uploaded_at(self, case_id: str) -> None:
+        """Update timestamp for cases which uploaded successfully"""
+        case_obj: models.Family = self.status_db.family(case_id)
+        case_obj.analyses[0].uploaded_at = dt.datetime.now()
+        self.status_db.commit()
```

Comment:
These does not seem to have anything to do with fohm. Should they not be somewhere closer to what the modulate like store for general reuse when wanting to update these fields?

There is for instance this `update_analysis_uploaded_at_date` in the nipt upload and I am sure several other ones in the other upload modules.